Automattic/wp-calypso
# media library: get items from redux store

#### Changes proposed in this Pull Request

- media library: use `getMediaSortedByDate` instead of `MediaListStore.getAll`
- add 'clearSite' to mimic flux media store behaviour (fresh list on e.g. filter change)

#### Testing instructions

Test the following on simple sites, AT and use both the classic post editor as well as Gutenberg.

1. Open the media library on `/media` and make sure all media items load (also do this on a site with >50 items)
2. Switch between filters _Images_, _Documents_, etc and make sure behaviour is the same as on prod
3. Switch to Google Photos and Pexels and make sure both load as well (for Pexels perform a search)
4. Copy items from 3. to your media library and make sure they immediately appear with a spinner and are added correctly and are persisted (use a browser refresh to check)
5. Edit a media item (flip, crop, rotate) and make sure changes are applied

Everything from 1. to 5. which can be applied should be done using the media modal as well (accessible from both editors).

related to #43663
